Jenkinsfile

```diff
@@ -200,18 +200,54 @@ pipeline {
                         stage("Run tests") {
                             steps {
                                 sh '''
-                                docker run --rm -t --privileged \
-                                  -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
-                                  -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
-                                  --name docker-pr$BUILD_NUMBER \
-                                  -e DOCKER_GITCOMMIT=${GIT_COMMIT} \
-                                  -e DOCKER_GRAPHDRIVER \
-                                  docker:${GIT_COMMIT} \
-                                  hack/make.sh \
-                                    binary-daemon \
-                                    dynbinary-daemon \
-                                    test-integration-flaky \
-                                    test-integration \
+                                rm -rf "$WORKSPACE/bundles"
+
+                                export CONTAINER_NAME=docker-pr$BUILD_NUMBER
+                                export TEST_INTEGRATION_DEST
+                                export TEST_SKIP_INTEGRATION
+                                export TEST_SKIP_INTEGRATION_CLI
+                                export TESTDEBUG
+                                export TESTFLAGS
+
+                                run_tests() {
+                                        [ -n "$TESTDEBUG" ] && rm= || rm=--rm;
+                                        docker run $rm -t --privileged \
+                                          -v "$WORKSPACE/bundles:/go/src/github.com/docker/docker/bundles" \
+                                          -v "$WORKSPACE/.git:/go/src/github.com/docker/docker/.git" \
+                                          --name "$CONTAINER_NAME" \
+                                          -e KEEPBUNDLE=1 \
+                                          -e TESTDEBUG \
+                                          -e TESTFLAGS \
+                                          -e TEST_INTEGRATION_DEST \
+                                          -e TEST_SKIP_INTEGRATION \
+                                          -e TEST_SKIP_INTEGRATION_CLI \
+                                          -e DOCKER_GITCOMMIT=${GIT_COMMIT} \
+                                          -e DOCKER_GRAPHDRIVER \
+                                          docker:${GIT_COMMIT} \
+                                          hack/make.sh \
+                                            dynbinary-daemon \
```

Comment:
looks like this `dynbinary-daemon` will be called 3 times

Reply:
having this cleaned up can be cleaned up in a follow up PR

Reply:
Done in this one